src/controllers/personsPublicRegisterNameController.ts

```diff
@@ -0,0 +1,78 @@
+import { NextFunction, Request, Response } from "express";
+import * as config from "../config";
+import { validationResult } from "express-validator";
+import { formatValidationError, getPageProperties } from "../validations/validation";
+import { BASE_URL, CHECK_YOUR_ANSWERS, PERSONS_NAME_ON_PUBLIC_REGISTER, PERSONAL_CODE } from "../types/pageURL";
+import { Session } from "@companieshouse/node-session-handler";
+import { saveDataInSession } from "../utils/sessionHelper";
+import { ClientData } from "../model/ClientData";
+import { getPreviousPageUrl } from "../services/url";
+import { USER_DATA, MATOMO_BUTTON_CLICK, PREVIOUS_PAGE_URL } from "../utils/constants";
+import { selectLang, addLangToUrl, getLocalesService, getLocaleInfo } from "../utils/localise";
+
+export const get = (req: Request, res: Response, next: NextFunction) => {
+    const lang = selectLang(req.query.lang);
+    const locales = getLocalesService();
+    const session: Session = req.session as any as Session;
+    const clientData: ClientData = session.getExtraData(USER_DATA) ? session.getExtraData(USER_DATA)! : {};
```

Comment:
we should not need to handle the null client data like this as it should never be null on this page.
it should be const clientData: ClientData = session.getExtraData(USER_DATA)!